server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2336,6 +2336,7 @@ private IndexWriterConfig getIndexWriterConfig() {
             iwc.setMaxFullFlushMergeWaitMillis(0);
         }
 
+        iwc.setCheckPendingFlushUpdate(config().getIndexSettings().isCheckPendingFlushEnabled());
```

Comment:
@CaptainDredge as we were discussing, lets create a separate issue to see if we can allow such settings to be dynamic by the use of IndexEventListener implementation which can allow for shard level writer config to be updated when index settings are updated.